src/ResourceManagement/CosmosDB/CassandraKeyspaceImpl.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Management.CosmosDB.Fluent
+{
+    using Microsoft.Azure.Management.CosmosDB.Fluent.Models;
+    using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal partial class CassandraKeyspaceImpl :
+        ExternalChildResource<ICassandraKeyspace,
+            CassandraKeyspaceGetResultsInner,
+            ICosmosDBAccount,
+            CosmosDBAccountImpl>,
+        ICassandraKeyspace,
+        CassandraKeyspace.Definition.IDefinition<CosmosDBAccount.Definition.IWithCreate>,
+        CassandraKeyspace.Update.IUpdate,
+        CassandraKeyspace.Definition.IDefinition<CosmosDBAccount.Update.IWithOptionals>
+    {
+        private ICassandraResourcesOperations Client { get { return Parent.Manager.Inner.CassandraResources; } }
+        private CassandraKeyspaceCreateUpdateParameters createUpdateParameters;
+        private ThroughputSettingsUpdateParameters throughputSettingsToUpdate;
+        private CassandraTablesImpl cassandraTables;
+
+        internal CassandraKeyspaceImpl(string name, CosmosDBAccountImpl parent, CassandraKeyspaceGetResultsInner inner)
+            : base(name, parent, inner)
+        {
+            this.cassandraTables = new CassandraTablesImpl(this);
+            SetCreateUpdateParameters();
+        }
+
+        private void SetCreateUpdateParameters()
+        {
+            this.createUpdateParameters = new CassandraKeyspaceCreateUpdateParameters()
+            {
+                Location = Parent.Inner.Location,
+                Resource = new CassandraKeyspaceResource(id: this.Name()),
+                Options = new Dictionary<string, string>(),
+            };
+        }
+
+        public string CassandraKeyspaceId()
+        {
+            return this.Inner.Resource.Id;
+        }
+
+        public string _rid()
+        {
+            return this.Inner.Resource._rid;
+        }
+
+        public object _ts()
+        {
+            return this.Inner.Resource._ts;
+        }
+
+        public string _etag()
+        {
+            return this.Inner.Resource._etag;
+        }
+
+        public string Id()
+        {
+            return this.Inner.Id;
+        }
+
+        public CassandraKeyspaceImpl WithOption(string key, string value)
+        {
+            this.createUpdateParameters.Options.Add(key, value);
+            return this;
+        }
+
+        public CassandraKeyspaceImpl WithOptionsAppend(IDictionary<string, string> options)
```

Comment:
What the two different?

Reply:
I have added these two function in all container, one is appending all container items, the other is replacing the current container (use clear and append according to weidong's comment).